src/python/pants/backend/python/subsystems/pytest.py

```diff
@@ -12,28 +12,65 @@ class PyTest(Subsystem):
   @classmethod
   def register_options(cls, register):
     super().register_options(register)
-    # TODO: This is currently bounded below `3.7` due to #6282.
+    register('--version', default='pytest>=3.0.7,<3.7', help="Requirement string for Pytest.")
+    register(
+      '--pytest-plugins',
+      type=list,
+      default=[
+        'pytest-timeout>=1.2,<1.3',
+        'pytest-cov>=2.4,<2.5',
+        "unittest2>=0.6.0,<=1.9.0 ; python_version<'3'"
+      ],
+      help="Requirement strings for any plugins or additional requirements you'd like to use.",
+    )
     register('--requirements', advanced=True, default='pytest>=3.0.7,<3.7',
-             help='Requirements string for the pytest library.')
+             help='Requirements string for the pytest library.',
+             removal_version="1.25.0.dev0", removal_hint="Use --version instead.")
     register('--timeout-requirements', advanced=True, default='pytest-timeout>=1.2,<1.3',
-             help='Requirements string for the pytest-timeout library.')
+             help='Requirements string for the pytest-timeout library.',
+             removal_version="1.25.0.dev0", removal_hint="Use --pytest-plugins instead.")
```

Comment:
Is there a good reason not to just have a single monolithic list to put all requirements in? It looks like we're just merging them anyway.

Reply:
Benjy and I talked about that. I argued for having `--version` be distinct from `--pytest-plugins` due to the ergonomics of modifying a list. Let's say a user likes our default plugins but wants a newer version of Pytest. They couldn't easily swap out the old Pytest version for the new one, so they'd have to end up doing this:

```ini
[pytest]
requirements: ['pytest>5.7', 'pytest-cov>=2.4,<2.5', 'pytest-cov>=2.4,<2.5']
```

Doing this would mean no more plugins:
```ini
[pytest]
requirements: ['pytest>5.7']
```

Instead, we allow for this:
```ini
[pytest]
version: "pytest>5.7"
pytest_plugins: +["pytest-flaky"]
```

--

Finally, what would we name a unified list? `requirements` is already sadly taken.